paasta_tools/utils.py

```diff
@@ -2852,6 +2852,12 @@ def load_service_instance_configs(
     return merged
 
 
+# be extremely sure you do not modify the return value of this.
```

Comment:
Hm, how much faster does this make things? service_configuration_lib already does some YAML caching:  https://github.com/Yelp/service_configuration_lib/blob/master/service_configuration_lib/__init__.py#L90

But it always deep-copies, which makes it a little slower. Just wondering if the speed is worth having to worry about accidental mutations here.

Reply:
Yeah, the deep copy is the second-slowest part. In my pyflame of stock `generate_deployments_for_service`, copy.deepcopy takes 15.5% of time, so roughly 25-30s. (`get_deploy_group_mappings` is the bulk of the rest of it)

Reply:
seems safe to do pickle+unpickle? should be faster iirc

Reply:
really? it would surprise me if serializing/deserializing is faster than deepcopy.